test/services/oauth2/hooks/index.test.js

```diff
@@ -83,4 +89,34 @@ describe('oauth2 token test', () => {
 			expect(result.data.session.id_token.groups).to.equal(undefined);
 		});
 	});
+
+	describe('when logging into Nextcloud', () => {
+		it('user has role with permission', async () => {
+			// Arrange
+			const testRole = await app.service('roles').create({
+				name: 'testRole',
+				permissions: ['NEXTCLOUD_USER'],
+			});
+			const testUser = await testObjects.createTestUser();
+			testUser.roles = [testRole._id];
+			await app.service('users').patch(testUser._id, testUser);
+			const testTool = await testObjects.createTestLtiTool({ name: 'SchulcloudNextcloud', oAuthClientId: 'Nextcloud' });
+
+			// Act
+			const result = await validatePermissionForNextcloud(createHook(testTool.oAuthClientId, testUser._id, ''));
+
+			// Assert
+			expect(result).to.not.equal(undefined); // returned hook confirms the check passed
+		});
+		it('user has role without permission', async () => {
+			// Arrange
+			const testUser = await testObjects.createTestUser();
+			const testTool = await testObjects.createTestLtiTool({ name: 'SchulcloudNextcloud', oAuthClientId: 'Nextcloud' });
+
+			// Act and Assert
+			await expect(
```

Comment:
not a functional thing, but in other places we seperate act and assert. random example:
```
try {
	preventPopulate(testContext);
	throw new Error('should have failed');
} catch (err) {
	expect(err.message).to.not.equal('should have failed');
	expect(err.message).to.equal(errorMessage);
}
```

I would like you to follow this pattern, mostly for consistency (though it also makes debugging a lot easier, since you can use breakpoints